programs/editor/bellabello.js

```diff
@@ -0,0 +1,498 @@
+                    var canvasContainerElement;
+
+                    editor = e;
+
+                    canvasContainerElement = editor.getCanvasContainerElement();
+                    savingOverlay = addStatusOverlay(canvasContainerElement, "document-save.png", 0);
+                    hasLocalUnsyncedOpsOverlay = addStatusOverlay(canvasContainerElement, "vcs-locally-modified.png", 0);
+                    disconnectedOverlay = addStatusOverlay(canvasContainerElement, "network-disconnect.png", 1);
+
+                    editor.addEventListener(Wodo.EVENT_BEFORESAVETOFILE, function() {
+                        savingOverlay.style.display = "";
+                    });
+                    editor.addEventListener(Wodo.EVENT_SAVEDTOFILE, function() {
+                        savingOverlay.style.display = "none";
+                    });
+                    editor.addEventListener(Wodo.EVENT_HASLOCALUNSYNCEDOPERATIONSCHANGED, function(has) {
+                        hasLocalUnsyncedOpsOverlay.style.display = has ? "" : "none";
+                    });
+                    editor.addEventListener(Wodo.EVENT_HASSESSIONHOSTCONNECTIONCHANGED, function(has) {
+                        disconnectedOverlay.style.display = has ? "none" : "";
+                    });
+                    editor.addEventListener(Wodo.EVENT_UNKNOWNERROR, handleEditingError);
+
+                    // load the document and get called back when it's live
+                    editor.joinSession(serverFactory.createSessionBackend(sessionId, memberId, server), onEditingStarted);
+                });
+            } else {
+                editor.joinSession(serverFactory.createSessionBackend(sessionId, memberId, server), onEditingStarted);
+            }
+        }, function(err) {
+            // TODO: handle error
+            runtime.log(err);
+        });
+    }
+
+    function loginSuccess(userData) {
+        runtime.log("connected:" + userData.full_name);
+        token = userData.securityToken || null;
+
+        if (args.sessionId) {
+            joinSession(args.sessionId);
+        }
+    }
+
+    window.jsGlobalInstance.registerUserData(args.username, args.userData);
+
+    // start the editor with network
+    runtime.log("starting collaborative editor for [jsglobal].");
+    require(["webodf/editor/backend/jsglobal/ServerFactory"], function (ServerFactory) {
+        serverFactory = new ServerFactory();
+        server = serverFactory.createServer();
+        // wait for a network connection to establish.
+        // call the callback when done, when finally failed, or
+        // when a timeout reached.
+        // the parameter to the callback is a string with the possible
+        // values: "unavailable", "timeout", "ready"
+        server.connect(8000, function (state) {
+            if (state === "ready") {
+                server.login(args.username, loginSuccess);
+            }
+        });
+    });
+}
+
+window.JsGlobalServer = (function() {
```

Comment:
Problem might be also that currently the plans/ideas with splitscreeneditor are not clear/agreed with everyone :) I see it as a nice demo and some example code, not more. And also something that needs finishing still, as the multiple sessions feature is yet to be completed.

My personal desires with bellabello is to have something that can be used as (automated) fuzzy testing tool for OT/syncing on the browser level, so one optionally also can interactively analyze things, other than with qtjsruntime.

Surely both things could/should share as much code as possible. It already does to large degrees, by the usual reusable modules.
But my first goal here is not to make bellabello only work by just adding a bella plugin to splitscreeneditor :) even if that would be technically nice. Rather would I like some tool designed to the one purpose and not more. E.g. do I currently not see the need to enrich  the testing environment with multiple sessions options, rather would I like to keep the testing env as simple as possible, for less code that might disturb the testing.

So let's align: what would you like us to achieve with splitscreeneditor itself?